ui/src/app/store/zone/types/base.ts

```diff
@@ -12,4 +15,41 @@ export type Zone = Model & {
   updated: string;
 };
 
-export type ZoneState = GenericState<Zone, APIError>;
+export type ZonePK = Zone[typeof ZONE_PK];
+
+export type ZoneProcess = {
+  processing: ZonePK[];
+  success: ZonePK[];
+};
+
+export type ZoneProcesses = {
+  [ZONE_ACTIONS.delete]: ZoneProcess;
+  [ZONE_ACTIONS.update]: ZoneProcess;
+};
+
+export type ZoneAPIMeta = {
+  formId: string | null;
+  modelPk: ZonePK | null;
+};
+
+export type ZoneAPISuccess<R> = {
+  result: R;
+};
+
+export type ZoneAPIError = ZoneAPIMeta & {
+  error: APIError;
+};
+
+export type ZoneStateError = ZoneAPIError & {
+  action: ValueOf<typeof ZONE_ACTIONS>;
+};
+
+export type ZoneState = {
+  errors: ZoneStateError[];
```

Comment:
I don't think we need two separate keys for `errors` and `eventErrors` like we have in `machine`. IMO we should just have one, and it should have all the info we need.